lib/integrations/awsoidc/deploydbservice.go

```diff
@@ -0,0 +1,599 @@
+//
+// # Pre-requirement: TaskRole creation
+//
+// The req.TaskRoleARN Role must have the following policy:
+//
+//		{
+//		    "Version": "2012-10-17",
+//		    "Statement": [
+//		        {
+//		            "Effect": "Allow",
+//		            "Action": [
+//		                "iam:DeleteRolePolicy",
+//		                "iam:PutRolePolicy",
+//		                "iam:GetRolePolicy"
+//		            ],
+//		            "Resource": "arn:aws:iam::123456789012:role/<req.TaskRoleARN>"
+//		        },
+//	         {
+//	             "Effect": "Allow",
+//	             "Action": [
+//	                  "rds:DescribeDBInstances",
+//	                  "rds:ModifyDBInstance"
+//	              ],
+//	              "Resource": "*"
+//	         },
+//		        {
+//		            "Effect": "Allow",
+//		            "Action": "logs:*",
+//		            "Resource": "*"
+//		        }
+//		    ]
+//		}
+//
+// And the following Trust Policy
+//
+//	{
+//	    "Version": "2012-10-17",
+//	    "Statement": [
+//	        {
+//	            "Effect": "Allow",
+//	            "Principal": {
+//	                "Service": "ecs-tasks.amazonaws.com"
+//	            },
+//	            "Action": "sts:AssumeRole"
+//	        }
+//	    ]
+//	}
+//
+// # Pre-requirement: AWS OIDC Integration Role
+//
+// To deploy those services the AWS OIDC Integration Role requires the following policy:
+//
+//	{
+//	    "Version": "2012-10-17",
+//	    "Statement": [
+//	        {
+//	            "Effect": "Allow",
+//	            "Action": [
+//	                "ecs:CreateCluster",
+//	                "ecs:PutClusterCapacityProviders",
+//	                "ecs:DescribeClusters",
+//	                "ecs:RegisterTaskDefinition",
+//	                "ecs:CreateService",
+//	                "ecs:DescribeServices",
+//	                "ecs:DeleteService",
+//	                "ecs:UpdateService"
+//	            ],
+//	            "Resource": "*"
+//	        },
+//	        {
+//	            "Effect": "Allow",
+//	            "Action": [
+//	                "iam:PassRole"
+//	            ],
+//	            "Resource": "arn:aws:iam::123456789012:role/<req.TaskRoleARN>"
+//	        }
+//	    ]
+//	}
+//
+// # Discovery and Database Service
+//
+// The Database Service will match on the suggested labels received in the request.
+func DeployDBService(ctx context.Context, clt DeployDBServiceClient, req DeployDBServiceRequest) (*DeployDBServiceResponse, error) {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	teleportConfigString, err := generateTeleportConfigString(req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	taskDefinition, err := upsertTask(ctx, clt, req, teleportConfigString)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	taskDefinitionARN := *taskDefinition.TaskDefinitionArn
+
+	cluster, err := upsertCluster(ctx, clt, *req.ClusterName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
I don't know anything about ecs, but creating the task before creating the cluster feels strange to me, doesn't the task run *on* the cluster? Maybe add comments explaining this ordering?